openquake/hazardlib/source/rupture.py

```diff
@@ -440,3 +441,148 @@ def get_cdppvalue(self, target, buf=1.0, delta=0.01, space=2.):
             cdpp[iloc] = dpp_target - mean_dpp
 
         return cdpp
+
+    def get_rupture_fraction_strikeslip(self, target, angle=False):
+        """
+        Obtain the directivity distance parameters for strike-slip defined by
+        Somerville et al., 1997, page 205.
+
+        :param target:
+            A mesh object representing the location of the target sites.
+        :param angle:
+            If ``True`` (by default), the rup_azimuth is calculated. If this
+            is set to ``False``, the rup_distance is calculated.
+        :returns:
+            rup_distance, a numpy array, represents the rupture fraction
+            distance to the target site.
+            rup_azimuth, a numpy array, represents the angle between the
+            rupture direction and the path to the site with respect to the
+            rupture (measured in degrees herein)
+        """
+        # check if the rupture is multi-patches
+        top_edge = self.surface.get_resampled_top_edge()
+        if len(top_edge) > 2:
+            raise ValueError(
+                'multi-patches rupture calculation has not yet been available')
+
+        idxs = self.surface.mesh.geodetic_min_distance(target, indices=True)
+        cls_lon = self.surface.mesh.lons.take(idxs)
+        cls_lat = self.surface.mesh.lats.take(idxs)
+        s_lon = target.lons
+        s_lat = target.lats
+        epi = Point(self.hypocenter.longitude, self.hypocenter.latitude)
+        p_pc = epi.point_at(1., 0., self.surface.get_strike())
```

Comment:
The names of the variables are not self-explanatory. I therefore suggest adding some comments to explain the logic of what you are doing.